### PR TITLE
#166114191 Finishing to create a view all loan repayment history 

### PR DESCRIPTION
**Pull request by Dom58**
# PR description Template

#### What does this PR do?
This pull request will help an admin and user to view all loan repayment history of the Quick Credit web application retrieved from  Postgres database
#### Description of Task to be completed?
Creation of view all loan repayment history endpoint, protect the endpoint from unauthorized using auth middleware
#### How should this be manually tested?
Clone the `ft-api-v2-view-all-loan-repayment-history-#166114191` branch to your local machine and navigate the `127.0.0.1:4000/api/v2/repayments/loans` after running the server using `npm run server` test it in postman
#### What are the relevant pivotal tracker stories?
#166114191 
#### Any background context you want to add?
You must  logging in to view this loan repayment history
#### Screenshots (if appropriate)
Responces

#### Questions:
No question

### CODE REVIEW

**Review by Dom58 — COMMENTED** [review]
Solved

**Review by Jkadhuwa — APPROVED** [review]
Looks fine bro.

**Comment by kodek-sleuth** [comment]
Looks fine

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules
 .env
 .nyc_output
-coverage
\ No newline at end of file
+coverage
+.vscode
```
Comment (meshack-mbuvi) [inline .gitignore:5]: Add newline at the end of every file

File: server/controllers/LoanController.js [hunk 2]
```diff
@@ -236,7 +236,24 @@ const loanController = {
     } else {
       statusMessageFunction(res, 400, `${loanStatus.badRequestMessage}` )
     }
-  }
-  
+  },
+  async allRepaymentLoan (req, res){
+    try{
+      const {rows} = await pool.query(queryTable.getAllRepayments)
+      if (rows.length === 0) {
```
Comment (meshack-mbuvi) [inline server/controllers/LoanController.js]: This can be improved to be `if(!rows.length)`

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: server/controllers/LoanController.js [hunk 3]
```diff
@@ -236,7 +236,24 @@ const loanController = {
     } else {
       statusMessageFunction(res, 400, `${loanStatus.badRequestMessage}` )
     }
-  }
-  
+  },
+  async allRepaymentLoan (req, res){
+    try{
+      const {rows} = await pool.query(queryTable.getAllRepayments)
```
Comment (meshack-mbuvi) [inline server/controllers/LoanController.js:272]: Who should be able to view the repayment history?

**Comment by meshack-mbuvi** [comment]
Update the PR to follow the standards already shared in the channel.

File: .gitignore (hunk 1 above)
Comment (Dom58) [reply .gitignore:5]: Thank you

File: server/controllers/LoanController.js (hunk 2 above)
Comment (Dom58) [reply server/controllers/LoanController.js]: admin should be able to view loan repayment and user should.

File: server/tests/acc-users.test.js [hunk 4]
```diff
@@ -1,347 +1,347 @@
-import chai from 'chai';
-import chaiHttp from 'chai-http';
-import jwt from 'jsonwebtoken';
-import dotenv from 'dotenv';
-import server from '../app.js'
+// import chai from 'chai';
+// import chaiHttp from 'chai-http';
+// import jwt from 'jsonwebtoken';
+// import dotenv from 'dotenv';
```
Comment (meshack-mbuvi) [inline server/tests/acc-users.test.js]: You have commented codes

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

File: server/tests/acc-users.test.js (hunk 4 above)
Comment (Dom58) [reply server/tests/acc-users.test.js]: Now I delete those files and commented codes. You may recheck review it again!

**Review by meshack-mbuvi — APPROVED** [review]

File: server/controllers/LoanController.js (hunk 3 above)
Comment (Dom58) [reply server/controllers/LoanController.js:272]: Admin and user !?

**Pull request merged**